### ISSUE TITLE
Record/keep extension source for later reuse (`quarto update extension ...`)

### ISSUE THREAD

**Issue by mcanouil** [quarto-dev/quarto-cli#11468]
Currently, when installation an extension from GitHub, if the extension name does not match the repository name, it's not possible to use `quarto update extension ...`.

The majority of extensions uses `quarto-<extension name>` as a repository name instead of `<extension name>`.
For example:

1. Install extensions:
   ```sh
   quarto add mcanouil/quarto-iconify
   quarto add quarto-ext/typst-templates/poster
   ```

2. Check the installed extensions

   ```sh
   quarto list extensions
   ```

   ```txt
   Id                         Version    Contributes        
   quarto-ext/poster          1.0.0      formats            
   mcanouil/iconify           2.1.0      shortcodes         
   ```

3. Update the extensions

   ```sh
   quarto update extension quarto-ext/poster
   ```

   ```txt
   Extension not found in local or remote sources
   ```

This means that it's not possible to update the extension `quarto-ext/poster` because the repository name (directory) is `typst-templates/poster` and not `poster`.
Same goes for the extension `mcanouil/iconify`.

This could be solved by recording/keeping the repository name (source) somewhere.
It could be in `_extension.yml` under `source` key.

Keeping the source would allow better extension management and easier updates, e.g., `quarto update extension all`.

**Comment by cderv** [comment]
Just a note for clarity that `quarto update extension` does work, but you need to know the github repo name 
````
> quarto update extension quarto-ext/quarto-poster
````

and we don't record it for now and we do store the extension in `org-name/ext-name`, `ext-name` being the registered name from the metadata of the extension. This name is also the one to use inside the document to load a filter for example. `filter: [quarto-ext/poster]`

So we need to 

- Find where to store the source information, and maybe other information. 
	We could adopt an approach similar to R package that does modify the DESCRIPTION in the installed package folder. We could write some fields in the `_extension.yml` with install datetime, type of extension (local zip, remote, ...) and source of the remote if so. 

- Show this information in `quarto list extension` indeed

- Adapt `quarto update` to use the source when installed extension is used (`quarto update quarto-ext/poster`) and probably allow to update from source directly `quarto update extension quarto-ext/typst-template/poster`

Thanks a lot for the report @mcanouil !

**Comment by gadenbuie** [comment]
I'm going to second writing the source info into `_extension.yml` on install. Relatedly, it'd also be nice if that could support a command to help users open extension docs (it's offered during install but I don't think there's a way to return to the docs if you select No at that point).

(Sidenote: there's enough here that it'd be nice to have a `quarto extension` subcommand. I never feel secure that I'm doing the right thing when I use `quarto add`, `quarto list` or `quarto update` to interact with extensions. `quarto extension add` or `quarto extension update` are more verbose but much more certain.)

**Comment by mcanouil** [comment]
Writing the source in `_extension.yml` is also what I had in mind.
I am a bit working on this for my VSCode extension (https://github.com/mcanouil/quarto-wizard), especially to provide an interface to update/install/remove/get help on extensions.

If this design is validated and if that's ok, I volunteer to work on this on my spare time and make a PR.

**Comment by cscheid** [comment]
We have a 2025 goal to improve third-party extensibility in Quarto, and rethinking our extension tooling definitely fits. `quarto extension` doesn't actually fit most of our other commands; from memory, I think they're all verbs.

That doesn't mean we shouldn't do it, but it means that we need to think about it more carefully.

**Comment by cderv** [comment]
> quarto extension doesn't actually fit most of our other commands; from memory, I think they're all verbs.

yes we have only `quarto tools` as left over. Previously it was `quarto tools install` but it has been switched (d2447e09292d9afdde579e0174ad5c8683bd844d) to verb with `quarto install` for all extensions or tools, and then `quarto add` was added as _better_ way for extension `quarto install extension <extension>` is now `quarto add <extension>`

So if we are changing this, we definitely need to see the current state of all this and make the clearer API that will last 😄

**Comment by mcanouil** [comment]
FYI, Quarto Wizard adds a `source: owner/repo` or `source: owner/repo@tag` upon install/update to the extension manifest.
